slp_tf/slp_tf/tfplan/transformers/tfplan_dataflow_creator.py

```diff
@@ -50,31 +39,42 @@ def __init__(self, otm: TfplanOTM, graph: DiGraph):
             graph=self.graph
         )
 
-        self.components_sgs: Dict[str, List[TfplanComponent]] = {}
-        self.sgs_relationships: Dict[str, List[str]] = {}
-
     def transform(self):
-        self.components_sgs = self.__match_components_and_sgs()
-        if not self.components_sgs:
-            return
+        self.dataflows.extend(self.__create_dataflows_from_straight_relationships())
+        self.dataflows.extend(self.__create_dataflows_from_sgs())
 
-        self.sgs_relationships = self.__find_sgs_relationships()
-        if not self.sgs_relationships:
-            return
+    def __create_dataflows_from_straight_relationships(self):
+        dataflows = []
 
-        self.dataflows.extend(self.__create_dataflows_from_sgs())
+        for component in self.components:
+            for related_component in self.components:
+                if component == related_component or self.__are_hierarchically_related(component, related_component):
+                    continue
+
+                if self.relationships_extractor.exist_valid_path(component.tf_resource_id,
```

Comment:
this duplicates the logic? the elif will execute as array X is executed against same array?
[a,b,c] will execute [a,c] and [c,a]